Example/Example/ExampleViewController.swift

```diff
@@ -9,42 +9,74 @@
 import UIKit
 import Ably
 
+var client: ARTRealtime!
+
 class ExampleViewController: UIViewController {
 
-    private var client: ARTRealtime!
+    @IBOutlet weak var usernameText: UITextField!
 
     override func viewDidLoad() {
         super.viewDidLoad()
+    }
 
+    /* Performs the login action */
+    @IBAction func loginAction(sender: AnyObject) {
         connectToAbly()
     }
 
     private func connectToAbly() {
         /* Set up a Realtime client that authenticates with the local web server auth endpoint */
         let options = ARTClientOptions()
-        options.authUrl = NSURL(string: "http://localhost:4567/auth")
+        /* Check if user provided with a login */
```

Comment:
As with previous comment, by using `authCallback` instead the beauty of this is that the `authCallack` can obtain a token at run time and this this bit of code does not depend on whether the user has a clientId yet or not.